lib/task_listener.js

```diff
@@ -164,7 +168,8 @@ export default class TaskListener extends EventEmitter {
     if (exceedsThreshold) return;
 
     let claims = await this.taskQueue.claimWork(availableCapacity);
-    claims.forEach(this.runTask.bind(this));
+    let tasksets = await Promise.all(claims.map(this.coalesceClaim.bind(this)));
+    await Promise.all(tasksets.map(this.runTaskset.bind(this)));
```

Comment:
Wouldn't this block claiming any more work until all calls to runTaskset (meaning the entire task execution) completes?